packages/providers/upload-aws-s3/src/index.ts

```diff
@@ -23,146 +36,158 @@ interface File {
   buffer?: Buffer;
 }
 
-// TODO V5: Migrate to aws-sdk v3
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-require('aws-sdk/lib/maintenance_mode_message').suppress = true;
-
-function hasUrlProtocol(url: string) {
-  // Regex to test protocol like "http://", "https://"
-  return /^\w*:\/\//.test(url);
-}
+type UploadCommandOutput = (
+  | CompleteMultipartUploadCommandOutput
+  | AbortMultipartUploadCommandOutput
+) & {
+  Location: string;
+};
 
-interface InitOptions extends Partial<AWS.S3.ClientConfiguration> {
+interface InitOptions {
   baseUrl?: string;
   rootPath?: string;
-  s3Options: AWS.S3.ClientConfiguration & {
+  s3Options: S3ClientConfig & {
+    // TODO Remove this in V5
+    accessKeyId?: AwsCredentialIdentity['accessKeyId'];
+    secretAccessKey?: AwsCredentialIdentity['secretAccessKey'];
+    // Keep this for V5
+    credentials?: AwsCredentialIdentity;
     params: {
       Bucket: string; // making it required
-      ACL?: string;
-      signedUrlExpires?: string;
+      ACL?: ObjectCannedACL;
+      signedUrlExpires?: number;
     };
   };
+  [k: string]: any;
 }
 
+const assertUrlProtocol = (url: string) => {
+  // Regex to test protocol like "http://", "https://"
+  return /^\w*:\/\//.test(url);
+};
+
+// TODO Remove this in V5 since we will only support the new config structure
+const extractCredentials = (options: InitOptions): AwsCredentialIdentity => {
+  // legacy
+  if ((options.accessKeyId, options.secretAccessKey)) {
```

Comment:
```suggestion
  if (options.accessKeyId && options.secretAccessKey) {
```
Is this what is intended here? I think as written it's equivalent to `if(options.secretAccessKey)`